### DISCUSSION TITLE
Have we considered leveraging vitest and vite for the community's frontend dev and build tooling?

### DISCUSSION THREAD

**Discussion by staccato-em** [saleor/saleor#10526] [General]
I want to offer to PoC [`vite`](https://vitejs.dev/) and [`vitest`](https://vitest.dev/) tools for local development and build tool support of the [saleor-dashboard](https://github.com/saleor/saleor-dashboard) project.

Can I ask for some feedback about interest level from the community and feedback from those familiar with the requirements of the build tooling to share context relevant to the feasibility of such a PoC?

I want to say that I'm _very_ pleased with/impressed by the quality of this ecosystem and I'm exploring this as a potential solution to support its continued quality.

Appreciate anybody willing to share their thoughts.

**Comment by timuric** [comment]
@staccato-em could you please share what motivates you to use `vite` for dashboard? What kind of DX advantages you think it could bring?

**Reply by staccato-em** [reply]
Part of it is opinion: DX in `vite` has always felt like a reduction in friction for achieving my typical dev/build pipelines:
- hmr
- static site generation
- tree shaking
- style asset compilation
All of that with the high performance of esm-based esbuild dev loops provides a nice performance experience.
[Reference](https://vitejs.dev/guide/why.html#the-problems)

I am aware of the use of webpack's esbuild plugin in the saleor dashboard tooling so it _may_ be a moot point in terms of pure dev-loop/build speed concerns.
